packages/@sanity/base/src/components/Document.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {useMemo} from 'react'
+import generateScriptLoader from '../util/generateScriptLoader'
+import uncaughtErrorHandler from '../util/uncaughtErrorHandler'
+import AppLoadingScreen from './AppLoadingScreen'
+import NoJavascript from './NoJavascript'
+
+export interface DocumentAsset {
+  path: string
+  hash?: string
+}
+
+export interface DocumentProps {
+  basePath?: string
+  charset?: string
+  title?: string
+  viewport?: string
+  loading?: React.ReactNode
+  staticPath?: string
+  favicons?: DocumentAsset[]
+  stylesheets?: DocumentAsset[]
+  scripts?: DocumentAsset[]
+}
+
+const DEFAULT_FAVICONS: DocumentAsset[] = [{path: 'favicon.ico'}]
+
+export default function Document(props: DocumentProps) {
+  const {
+    basePath: basePathProp = '',
+    charset = 'utf-8',
+    title = 'Sanity',
+    viewport = 'width=device-width, initial-scale=1, viewport-fit=cover',
+    loading = 'Connecting to Sanity.io',
+    staticPath: staticPathProp = '/static',
+    favicons: faviconsProp = DEFAULT_FAVICONS,
+    stylesheets: stylesheetsProp = [],
+    scripts: scriptsProp = [],
```

Comment:
Won't this invalidate the useMemos that has these props as one of their dependencies when these props aren't given? Could perhaps consider re-using the same empty array instance.